src/main/java/com/metamx/emitter/core/HttpPostEmitter.java

```diff
@@ -374,6 +383,45 @@ public void run()
     }
 
     /**
+     * Serialize and compress (gzip) messages into a batch.
+     *
+     * @param messages list of JSON objects, one per message
+     * @return serialized and compressed JSON array
+     */
+    protected byte[] serializeAndCompressBatch(List<byte[]> messages) throws IOException
+    {
+      final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      try (
+          final GZIPOutputStream gzos = new GZIPOutputStream(baos)
```

Comment:
Just as an FYI, we're OK for us, but some versions of java have a nasty bug in their `FilterOutputStream` impl

old code reads like this:

http://www.docjar.com/html/api/java/io/FilterOutputStream.java.html
```java
public void close() throws IOException {
  try {
    flush();
  } catch (IOException ignored) {
  }
  out.close();
}
```


but newer (fixed) code reads like this:

```java
    public void close() throws IOException {
        try (OutputStream ostream = out) {
            flush();
        }
    }
```

But stupid things like that (and other java bugs) are things we should have worked around in java-util.